src/solver/optimisation/opt_rename_problem.cpp

```diff
@@ -0,0 +1,364 @@
+                      const std::string& location,
+                      const std::string& timeIdentifier)
+{
+    std::string result = name + SEPARATOR + location + SEPARATOR + timeIdentifier;
+    std::replace(result.begin(), result.end(), ' ', '*');
+    return result;
+}
+
+void VariableNamer::SetLinkVariableName(unsigned int var, const std::string& variableType)
+{
+    const auto location = origin_ + AREA_SEP + destination_;
+    targetUpdater_.UpdateTargetAtIndex(
+      BuildName(variableType, LocationIdentifier(location, LINK), TimeIdentifier(timeStep_, HOUR)),
+      var);
+}
+
+void VariableNamer::SetAreaVariableName(unsigned int var, const std::string& variableType)
+{
+    targetUpdater_.UpdateTargetAtIndex(
+      BuildName(variableType, LocationIdentifier(area_, AREA), TimeIdentifier(timeStep_, HOUR)),
+      var);
+}
+
+void VariableNamer::SetAreaVariableName(unsigned int var, const std::string& variableType, int layerIndex)
+{
+    targetUpdater_.UpdateTargetAtIndex(BuildName(variableType,
+                                                 LocationIdentifier(area_, AREA) + SEPARATOR
+                                                   + "Layer<" + std::to_string(layerIndex) + ">",
+                                                 TimeIdentifier(timeStep_, HOUR)),
+                                       var);
+}
+
+void VariableNamer::SetThermalClusterVariableName(unsigned int var,
+                                                  const std::string& variableType,
+                                                  const std::string& clusterName)
+{
+    const auto location
+      = LocationIdentifier(area_, AREA) + SEPARATOR + "ThermalCluster" + "<" + clusterName + ">";
+
+    targetUpdater_.UpdateTargetAtIndex(
+      BuildName(variableType, location, TimeIdentifier(timeStep_, HOUR)), var);
+}
+
+void VariableNamer::DispatchableProduction(unsigned int var, const std::string& clusterName)
+{
+    SetThermalClusterVariableName(var, "DispatchableProduction", clusterName);
+}
+
+void VariableNamer::NODU(unsigned int var, const std::string& clusterName)
+{
+    SetThermalClusterVariableName(var, "NODU", clusterName);
+}
+
+void VariableNamer::NumberStoppingDispatchableUnits(unsigned int var, const std::string& clusterName)
+{
+    SetThermalClusterVariableName(var, "NumberStoppingDispatchableUnits", clusterName);
+}
+
+void VariableNamer::NumberStartingDispatchableUnits(unsigned int var, const std::string& clusterName)
+{
+    SetThermalClusterVariableName(var, "NumberStartingDispatchableUnits", clusterName);
+}
+
+void VariableNamer::NumberBreakingDownDispatchableUnits(unsigned int var, const std::string& clusterName)
+{
+    SetThermalClusterVariableName(var, "NumberBreakingDownDispatchableUnits", clusterName);
+}
+
+void VariableNamer::NTCDirect(unsigned int var, const std::string& origin, const std::string& destination)
+{
+    origin_ = origin;
+    destination_ = destination;
+    SetLinkVariableName(var, "NTCDirect");
```

Comment:
For consistency with the rest, you should pass `origin` and `destination` to `SetLinkVariableName`, here and in `VariableNamer::IntercoDirectCost`, `VariableNamer::IntercoIndirectCost`.